actions/docs-verifier/src/RedirectionVerifier/WhatsNewConfigurationReader.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.IO;
+using System.Text.Json;
+
+namespace RedirectionVerifier
+{
+    public static class WhatsNewConfigurationReader
+    {
+        private static readonly JsonSerializerOptions s_options = new() { AllowTrailingCommas = true };
+        private static WhatsNewConfiguration s_cachedWhatsNewConfiguration = null!;
+        private const string WhatsNewConfigurationFileName = ".whatsnew.json";
+
+        /// <summary>
+        /// Retrieves the configured "What's new" directory from <c>.whatsnew.json</c>.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Failed to read <c>.whatsnew.json</c>.</exception>
+        public static string? GetWhatsNewPath()
+        {
+            // If there are cached configuration values for "What's new", use 'em.
+            if (s_cachedWhatsNewConfiguration?.NavigationOptions?.WhatsNewPath is not null)
```

Comment:
If `NavigationOptions` is not null, but `WhatsNewPath` is null, we'll end up re-reading the file unnecessarily just to figure out that `WhatsNewPath` is null again. Correct me if I'm missing something.

```suggestion
            if (s_cachedWhatsNewConfiguration?.NavigationOptions is not null)
```